### ISSUE TITLE
ActOnEveryObject appears not to track resets correctly

### ISSUE THREAD

**Issue by tcolemanld** [reactiveui/ReactiveUI#1675]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

ActOnEveryObject does not correctly track if `This` collection is reset

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

```csharp
public class Foo : ReactiveObject {
	public Foo(int id) {
		Id = $"Foo{id}";
	}
	public string Id { get; }
}

[Fact]
public void ActOnEveryObject_WithReset() {
	using (TestUtils.WithScheduler(ImmediateScheduler.Instance)) {
		var original = new ReactiveList<Foo>();

		var copy = new ReactiveList<Foo>();
		original.ActOnEveryObject(
			onAdd: x => {
				_output.WriteLine($"Adding {x.Id} to copy");
				copy.Add(x);
			},
			onRemove: x => {
				_output.WriteLine($"Removing {x.Id} from copy");
				copy.Remove(x);
			});

		var foo1 = new Foo(1);
		var foo2 = new Foo(2);
		original.Add(foo1);
		original.Add(foo2);

		Assert.Equal(2, original.Count);
		Assert.Equal(2, copy.Count);

		using (original.SuppressChangeNotifications()) {
			original.Remove(foo2);
		}

		Assert.Equal(1, original.Count);
		Assert.Equal(1, copy.Count); // fails. actually 2.

		/* output is
			Adding Foo1 to copy
			Adding Foo2 to copy
			Removing Foo1 from copy       <- curious
			Adding Foo1 to copy
		*/
	}
}
```

**What is the expected behavior?**

`copy` should contain the same items as `original`

**What is the motivation / use case for changing the behavior?**

ActOnEveryObject docs say that reset is handled.

**Which versions of ReactiveUI, and which platform / OS are affected by this issue? Did this work in previous versions of ReativeUI? Please also test with the [latest stable](https://www.nuget.org/profiles/reactiveui) and [development snapshot](https://reactiveui.net/docs/getting-started/installation/nuget-packages/development-package-feed)**

Windows. ReactiveUI 8.3.1

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

I suppose the problem is either with ActOnEveryObject, or possibly with ReactiveList<> emitting the wrong notifications.

**Comment by Wouterdek** [comment]
Yes, I have noticed this bug too.
It is caused by the following:
 - `ActOnEveryObject()` relies on the `Changing` and `Changed` observables to handle resets (`Clear()`). On `Changing` reset, it acts as if everything will be removed and then on `Changed` reset (after the actual changes) everything that is in the list is added back. This deals efficiently with the problem that it is unknown which elements in the list are new, were removed or were not changed at all. ([Source code for `ActOnEveryObject()`](https://github.com/reactiveui/ReactiveUI/blob/50403240c792d6d870fbe74961d5df297b4a1f2e/src/ReactiveUI/AutoPersistHelper.cs#L187))
 - When `SuppressChangeNotifications()` is used, `Changing` and `Changed` are not triggered. ([Source code for `ClearItems()`](https://github.com/reactiveui/ReactiveUI/blob/50403240c792d6d870fbe74961d5df297b4a1f2e/src/ReactiveUI/ReactiveList.cs#L298))
 - Instead, nothing happens until the disposable returned by `SuppressChangeNotifications()` is disposed, then `publishResetNotification()` is called. [(Source code for `SuppressChangeNotifications()`)](https://github.com/reactiveui/ReactiveUI/blob/50403240c792d6d870fbe74961d5df297b4a1f2e/src/ReactiveUI/ReactiveList.cs#L580)
 - `publishResetNotification()` triggers `Changing` and `Changed` with a reset `NotifyCollectionChangedEventArgs` event. [Source code for `publishResetNotification()`](https://github.com/reactiveui/ReactiveUI/blob/50403240c792d6d870fbe74961d5df297b4a1f2e/src/ReactiveUI/ReactiveList.cs#L535)
 - **However, the `Changing` event is too late.** `ActOnEveryObject()` will try to call `onRemoved` on all elements in the list, assuming they will be removed after the event. But they were already removed before, during the time `SuppressChangeNotifications()` was used. This means `ActOnEveryObject()` has no way of knowing what items were removed.

I'm not quite sure if this would break anything or work at all, but perhaps this could be fixed by pushing the reset `Changing` notification in `SuppressChangeNotifications()` instead of in the disposable that is called after the changes.
